Sources/Utility/File.swift

```diff
@@ -15,59 +15,42 @@ public enum FopenMode: String {
     case write = "w"
 }
 
-public func fopen(_ path: String, mode: FopenMode = .read) throws -> NSFileHandle {
-    let handle: NSFileHandle!
+public func fopen(_ path: String, mode: FopenMode = .read) throws -> FileHandle {
+    let handle: FileHandle!
     switch mode {
-    case .read: handle = NSFileHandle(forReadingAtPath: path)
+    case .read: handle = FileHandle(forReadingAtPath: path)
     case .write:
-        #if os(OSX) || os(iOS)
-            guard NSFileManager.`default`().createFile(atPath: path, contents: nil) else {
-                throw Error.couldNotCreateFile(path: path)
-            }
-        #else
-            guard NSFileManager.defaultManager().createFile(atPath: path, contents: nil) else {
-                throw Error.couldNotCreateFile(path: path)
-            }
-        #endif
-        handle = NSFileHandle(forWritingAtPath: path)
+        guard FileManager.`default`().createFile(atPath: path, contents: nil) else {
+            throw Error.couldNotCreateFile(path: path)
+        }
+        handle = FileHandle(forWritingAtPath: path)
     }
     guard handle != nil else {
         throw Error.fileDoesNotExist(path: path)
     }
     return handle
 }
 
-public func fopen<T>(_ path: String..., mode: FopenMode = .read, body: (NSFileHandle) throws -> T) throws -> T {
+public func fopen<T>(_ path: String..., mode: FopenMode = .read, body: (FileHandle) throws -> T) throws -> T {
     let fp = try fopen(Path.join(path), mode: mode)
     defer { fp.closeFile() }
     return try body(fp)
 }
 
-public func fputs(_ string: String, _ handle: NSFileHandle) throws {
+public func fputs(_ string: String, _ handle: FileHandle) throws {
     guard let data = string.data(using: NSUTF8StringEncoding) else {
         throw Error.unicodeEncodingError
     }
 
-    #if os(OSX) || os(iOS)
-        handle.write(data)
-    #else
-        handle.writeData(data)
-    #endif
+    handle.write(data)
 }
 
-public func fputs(_ bytes: [UInt8], _ handle: NSFileHandle) throws {
-    var bytes = bytes
-    let data = NSData(bytes: &bytes, length: bytes.count)
-
-    #if os(OSX) || os(iOS)
-        handle.write(data)
-    #else
-        handle.writeData(data)
-    #endif
+public func fputs(_ bytes: [UInt8], _ handle: FileHandle) throws {
+    handle.write(Data(bytes: bytes))
```

Comment:
Not a criticism but just a question: do you know if the compiler is smart enough to optimize out the copying of the byte array here since it knows that it's creating a temporary Data?